core/module.go

```diff
@@ -63,6 +63,9 @@ type Module struct {
 
 	// InstanceID is the ID of the initialized module.
 	InstanceID *call.ID
+
+	// Server is the dagql server
+	Server *dagql.Server
```

Comment:
This shouldn't go here.

I see they're attached so that `ModuleFunction` can use this - we should probably instead just add it as an arg everywhere it needs to go. It feels like a code smell to attach it in this way to me. I feel like it's tempting some very subtle bugs in the context of https://github.com/dagger/dagger/pull/7759.

Reply:
I agree but the function where `dag` is used is `Call`, which implement `Callable` and I cannot access `dag` everywhere, that's why I did this hack.
Do you have an idea to workaround that? Because adding `dag` in opts could also lead to other issues such as a nil dagql server... which can be catch but will lead to weird error no?

Reply:
Yup, adding it to the `Opts` sounds fine to me.

If the dagql server is `nil` we can error out to avoid panicking - yes, that's weird, so we should make sure to always pass it in. Where can't you access `dag`? Looking at all call-sites of `Call`, there's either a `dag` in the current scope, or you can easily propagate it down. For `Module.Initialize`, you can get it from `moduleSchema.dag`. For `objFun`, you need to propagate it through `functions`.

Reply:
Ok I made the changes for that, was pretty easy with your hints!